test/streamingReceiver.spec.ts

```diff
@@ -247,4 +249,220 @@ describe("ReceiveBatch from Queue/Subscription", function(): void {
 
     await receiveListener.stop();
   });
+
+  it("Abandoned message is retained in the Queue with incremented deliveryCount. After 10 times, you can only get it from the dead letter queue.", async function(): Promise<
+    void
+  > {
+    await queueClient.sendBatch(testMessages);
+
+    let checkDeliveryCount0 = 0;
+    let checkDeliveryCount1 = 0;
+
+    const receiveListener = await queueClient.receive(
+      (msg: ServiceBusMessage) => {
+        if (msg.messageId === testMessages[0].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount0);
+          checkDeliveryCount0++;
+        } else if (msg.messageId === testMessages[1].messageId) {
+          should.equal(msg.deliveryCount, checkDeliveryCount1);
+          checkDeliveryCount1++;
+        }
+        return msg.abandon();
+      },
+      (err: Error) => {
+        should.not.exist(err);
+      },
+      { autoComplete: false }
+    );
+
+    await delay(4000);
+
+    await receiveListener.stop();
+
+    should.equal(checkDeliveryCount0, maxDeliveryCount);
+    should.equal(checkDeliveryCount1, maxDeliveryCount);
+
+    await testPeekMsgsLength(queueClient, 0); // No messages in the queue
+
+    const deadLetterQueuePath = Namespace.getDeadLetterQueuePathForQueue(queueClient.name);
+    const deadletterQueueClient = namespace.createQueueClient(deadLetterQueuePath);
+
+    const deadLetterMsgs = await deadletterQueueClient.receiveBatch(2);
+    should.equal(Array.isArray(deadLetterMsgs), true);
+    should.equal(deadLetterMsgs.length, 2);
+    should.equal(deadLetterMsgs[0].deliveryCount, maxDeliveryCount);
+    should.equal(deadLetterMsgs[1].deliveryCount, maxDeliveryCount);
```

Comment:
Here lets also test the message id. Otherwise this test will pass if there were other messages in the dead letter queue with same deliver count

Reply:
Made commits for this. Working fine.
A small question : 
Is it possible that the two messages in the deadletter are in different order compared to the sent messages ?